identity.go

```diff
@@ -622,3 +625,86 @@ func loadCABundle(caAddr string) (*x509.CertPool, *CaPool, error) {
 		return pool, caPool, nil
 	}
 }
+
+func (id *ID) ValidFor(address string) error {
+	return ValidFor(id, address)
+}
+
+// Define base errors
+var (
+	ErrInvalidAddress            = errors.New("invalid address")
+	ErrInvalidAddressForIdentity = errors.New("identity is not valid for provided host")
+)
+
+// Define a struct for detailed errors
+type AddressError struct {
+	BaseErr  error
+	Host     string
+	ValidFor []string
+}
+
+// Implement the error interface
+func (e *AddressError) Error() string {
+	return fmt.Sprintf("%s: [%s]. is valid for: [%s]", e.BaseErr.Error(), e.Host, strings.Join(e.ValidFor, ", "))
+}
+
+// Implement Unwrap to work with errors.Is
+func (e *AddressError) Unwrap() error {
+	return e.BaseErr
+}
+
+// ValidFor checks if the identity is valid for the given address
+func ValidFor(id Identity, address string) error {
+	address = strings.TrimPrefix(address, "tls:")
```

Comment:
This should probably use url.Parse() so it will handle `http://` and `tls:` and any other "<proto>:[//]<hostname>:<port>[/<path>][?<querystring>]" format. Still can use the `.Host` on it to then split hostname port. The URL parsing won't do that for you.